llvm/include/llvm/Frontend/HLSL/RootSignatureMetadata.h

```diff
@@ -48,6 +48,93 @@ class RootSignatureValidationError
   }
 };
 
+class OffsetOverflowError : public ErrorInfo<OffsetOverflowError> {
```

Comment:
Nothing needed for this PR as you're just following the established pattern, but we should probably come back and take a look at this error heirarchy at some point - I'm not really convinced we need this many distinct errors given all we ever do with them is print the error message (as opposed to handling specific errors in some way). We might be able to simplify these to something only a bit more structured than a `StringError` without any real loss to functionality or readability.